media/com_fabrik/js/tipsBootStrapMock.js

```diff
@@ -86,6 +100,20 @@ define(['jquery', 'fab/fabrik'], function (jQuery, Fabrik) {
                     thisOpts.defaultPos = thisOpts.position;
                     delete(thisOpts.position);
                 }
+                if (thisOpts.setwidth) {
+                    thisOpts.maxwidth = thisOpts.setwidth;
+                    delete(thisOpts.setwidth);
+                } 
+                var headertxt = '';
+                if (thisOpts.tipuselabel == '1'){
+                    headertxt = this.innerText;                 
+                } else if (thisOpts.tiptitle === '') {
+                    headertxt = thisOpts.heading;                 
+                } else if (thisOpts.tiptitle != 'x') {
+                    headertxt = thisOpts.tiptitle;                 
+                }
+                thisOpts.heading = headertxt;
```

Comment:
Is there a reason for using a local variable `headertxt` rather than just assigning directly to `thisOpts.heading`?

Reply:
thisOpts.heading is set in the tipTextAndValidations function in element.php.
It will be blank unless there there is any validations (then it is 'Validation' ).
So I use headertxt to determine what will be used for thisOpts.heading.
It could be...
1. the title as entered in the configuration form
2. thisOpts.heading (from tipTextAndValidations function)
3. the element label (as determined if 'x' was entered as the title)

Reply:
My comment is purely stylistic - which means it is a subjective preference on my part.